scanpy/tools/_rank_genes_groups.py

```diff
@@ -35,6 +64,10 @@ def rank_genes_groups(
         The key of the observations grouping to consider.
     use_raw : `bool`, optional (default: `True`)
         Use `raw` attribute of `adata` if present.
+    use_weights : `bool`, optional (default: `False`)
+        Check either input matrix to `adata` has weights for observation or not.
+    weights: `list`, optional (default: 'None')
```

Comment:
“list”? elsewhere you do `weights.columns = ['Wt']`. Lists don’t have a “column” attribute.

I think users should be able to pass a 1D numpy array or a pandas Series or a list of weights that you then convert to whatever you want to use.

I also don’t understand the text: “Use `weights` vector as sample weights for input of `adata`.”

“Sample weights” makes sense, but what does “input of `adata`” mean? What are the weights used for?

Reply:
Previously , I was using list for  scanpy/tools/_rank_genes_groups.py and dataframe for scanpy/plotting/_anndata.py, bacause that was according to calculation of weighted means and weighted variance. But now i have changed input to a dataframe for both files and for farmer i process it to convert to list.

Reply:
A 1-column dataframe doesn’t make sense. Please allow users to pass any sequence of numbers (`list`, `numpy.ndarray`, `pandas.Series`).

You can simply convert that sequence into a dataframe inside of the function.